presto-main/src/test/java/com/facebook/presto/sql/planner/sanity/TestVerifyNoOriginalExpression.java

```diff
@@ -0,0 +1,298 @@
+        OrderingScheme orderingScheme = new OrderingScheme(
+                groupingKeys,
+                orderings);
+        ImmutableMap<VariableReferenceExpression, AggregationNode.Aggregation> aggregations = ImmutableMap.of(
+                variableReferenceExpression,
+                new AggregationNode.Aggregation(
+                        callExpression,
+                        Optional.of(callExpression),
+                        Optional.of(orderingScheme),
+                        false,
+                        Optional.of(new VariableReferenceExpression("orderkey", BIGINT))));
+        ImmutableSet<Integer> globalGroupingSets = ImmutableSet.of(1);
+        AggregationNode.GroupingSetDescriptor groupingSets = new AggregationNode.GroupingSetDescriptor(groupingKeys, groupingSetCount, globalGroupingSets);
+        ImmutableList<VariableReferenceExpression> preGroupedVariables = ImmutableList.of();
+        Optional<VariableReferenceExpression> hashVariable = Optional.of(variableReferenceExpression);
+        Optional<VariableReferenceExpression> groupIdVariable = Optional.of(variableReferenceExpression);
+        AggregationNode aggregationNode = new AggregationNode(
+                new PlanNodeId("1"),
+                valuesNode,
+                aggregations,
+                groupingSets,
+                preGroupedVariables,
+                AggregationNode.Step.SINGLE,
+                hashVariable,
+                groupIdVariable);
+        testValidation(aggregationNode);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testValidateForApplyFailed()
+    {
+        ImmutableMap<VariableReferenceExpression, RowExpression> map = ImmutableMap.of(variableReferenceExpression, castToRowExpression(new SymbolReference("count")));
+        Assignments assignments = new Assignments(map);
+        ImmutableList<VariableReferenceExpression> variableReferenceExpressions = ImmutableList.of(variableReferenceExpression);
+        ApplyNode applyNode = builder.apply(
+                assignments,
+                variableReferenceExpressions,
+                valuesNode,
+                valuesNode);
+        testValidation(applyNode);
+    }
+
+    @Test
+    public void testTableFinish()
+    {
+        TableWriterNode.WriterTarget writerTarget = new TableWriterNode.WriterTarget() {
+            @Override
+            public String toString()
+            {
+                return null;
+            }
+        };
+        TableFinishNode tableFinishNode = new TableFinishNode(
+                new PlanNodeId("1"),
+                valuesNode,
+                writerTarget,
+                variableReferenceExpression,
+                Optional.empty(),
+                Optional.empty());
+        testValidation(tableFinishNode);
+    }
+
+    @Test
+    public void TableWriter()
```

Comment:
`testTableWriter` or `testTableWriterNode`. (I might add Node to most of the test cases. But it's ok without as well. Up to you.